src/github.com/matrix-org/go-neb/services/circleci/circleci.go

```diff
@@ -0,0 +1,270 @@
+			//
+			// This is identical to the format of Slack Notifications for Travis-CI:
+			// https://docs.travis-ci.com/user/notifications#Customizing-slack-notifications
+			//
+			// As this is CircleCI it also supports all CircleCI fields
+			// Compare with https://circleci.com/docs/api/v1-reference/#build
+			//
+			// The following variables are available:
+			//   repository_slug: your Git* repo identifier (like svenfuchs/minimal)
+			//   reponame: the slug without the username
+			//   repository_name: the slug without the username //Deprecated for CircleCI use "reponame" instead
+			//   build_num: build number
+			//   build_number: build number //Deprecated for CircleCI use "build_num" instead
+			//   build_id: build id //Deprecated for CircleCI use "build_num" instead as this value doesn't really exist in CircleCI
+			//   branch: branch build name
+			//   commit: shortened commit SHA
+			//   committer_name: commit author name
+			//   author: commit author name //Deprecated for CircleCI use "committer_name" instead
+			//   body: commit message of build
+			//   commit_message: commit message of build //Deprecated for CircleCI use "body" instead
+			//   commit_subject: first line of the commit message
+			//   result: result of build
+			//   message: CircleCI message to the build
+			//   duration: total duration of all builds in the matrix
+			//   elapsed_time: time between build start and finish
+			//   build_url: URL of the build detail
+
+			Template string `json:"template"`
+		} `json:"repos"`
+	} `json:"rooms"`
+}
+
+// Converts a webhook notification into a map of template var name to value
+func notifToTemplate(n WebhookNotification) map[string]string {
+	t := make(map[string]string)
+	//Get Payload to variable
+	p := n.Payload
+	t["repository_slug"] = p.Username + "/" + p.Reponame
+	t["repository"] = t["repository_slug"] // Deprecated form but still used everywhere in people's templates
+	t["repository_name"] = p.Reponame
+	t["build_number"] = strconv.Itoa(p.BuildNum)
+	t["build_id"] = t["build_number"] // CircleCI doesn't have a difference between number and ID but to be consistent with TravisCI
+	shaLength := len(p.VcsRevision)
+	if shaLength > 10 {
+		shaLength = 10
+	}
+	t["commit"] = p.VcsRevision[:shaLength] // shortened commit SHA
+	t["author"] = p.CommitterName      // author: commit author name
+	// commit_message: commit message of build
+	// commit_subject: first line of the commit message
+	t["commit_message"] = p.Body
+	subjAndMsg := strings.SplitN(p.Body, "\n", 2)
+	t["commit_subject"] = subjAndMsg[0]
+	if p.Status != "" {
+		t["result"] = p.Status
+	}
+	t["message"] = p.Outcome // message: CircleCI message to the build
+
+	if !p.StartTime.IsZero()  && !p.StopTime.IsZero() {
+			t["duration"] = p.StopTime.Sub(p.StartTime).String()
+			t["elapsed_time"] = t["duration"]
+	}
+
+	t["build_url"] = p.BuildURL
+
+	// Make the full struct data into the Map without manual mapping
+	rawT := structs.Map(p)
```

Comment:
I'm not a fan of this *magic*
best to hardcode all the conversions if anything then for consistency

Reply:
(plus its what the travisci service does)

Reply:
Hm any Idea how I manualy should map the Steps Array?